packages/common/src/use-select/use-select.js

```diff
@@ -69,9 +69,16 @@ const useSelect = ({
   simpleValue,
   compareValues,
 }) => {
-  const [state, originalDispatch] = useReducer(reducer, { optionsTransformer, propsOptions }, init);
+  const [propsOptions, setPropsCache] = useState(initialOptions);
+  const [state, originalDispatch] = useReducer(reducer, { optionsTransformer, propsOptions: initialOptions }, init);
   const dispatch = (action) => originalDispatch({ ...action, optionsTransformer });
```

Comment:
We can pass the `compareValues` here so no need to pass it to all following dispatches